### PR TITLE
Cindy text

### PR DESCRIPTION
**Pull request by cindylee0113**
Edited how to search for the missing end tags because I left out different cases where it's alternating, or all end tags come last but cancel out the front start tags, if there's an equal number of tags but the tags don't match up in the order they appear in.....etc............ (please let me know if you think of more test cases....)

Also added a "Add Link" input area. So if the user wants to add a hyperlink, insert the url, press Add Link button -> wallah, the BBCode version is added to the input message box

### CODE REVIEW

File: src/main/java/codeu/controller/ChatServlet.java [hunk 1]
```diff
@@ -401,118 +401,57 @@ else if (messageArray[i].equals(":@")){
...
-    for (int i = 0; i < difference; i++) {
-      cleanedMessageContent += "</i>";
-    }
+        // if no more end or no start tags 
+        if (start_sym_tag == -1 && end_sym_tag == -1){
+            break;
+        }
+        
+        // if no start but it's a leftover end
+        // check to see if the end cancel out
```
Comment (aljonpineda) [inline src/main/java/codeu/controller/ChatServlet.java]: What do you mean by cancelling out?

Comment (cindylee0113) [reply src/main/java/codeu/controller/ChatServlet.java]: Cancel out as in if start_tag is 2, check to see if there are leftover [/b] let's say to cancel out the two start_tags

Comment (aljonpineda) [reply src/main/java/codeu/controller/ChatServlet.java]: So, for instance, do you mean something like: `[b] [b] [/b] [/b] text`?

Comment (cindylee0113) [reply src/main/java/codeu/controller/ChatServlet.java]: Yup!

**Pull request merged**